code/modules/food_and_drink/ingredients.dm

```diff
@@ -582,41 +582,45 @@
 	var/toppingstext = null
 
 	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W, /obj/item/reagent_containers/food/snacks/))
-			var/obj/item/reagent_containers/food/snacks/F = W
-			if(!F.custom_food)
-				return
+		if (istype(W, /obj/item/reagent_containers/food))
+			add_ingredient(W, user)
+		else
+			return
+	proc/add_ingredient(obj/item/W as obj, mob/user as mob)
+		var/obj/item/reagent_containers/food/snacks/F = W
+		if(!F.custom_food)
+			return
+		if(user)
 			boutput(user, "<span class='notice'>You add [W] to [src].</span>")
-			topping = 1
+		topping = 1
+		if(istype(F,/obj/item/reagent_containers/food/snacks))
 			food_effects += F.food_effects
 			src.AddComponent(/datum/component/consume/food_effects, src.food_effects)
-			if (F.real_name)
-				toppings += F.real_name
-			else
-				toppings += W.name
-			toppingstext = copytext(html_encode(english_list(toppings)), 1, 512)
-			name = "uncooked [toppingstext] pizza"
-			desc = "A pizza with [toppingstext] toppings. You need to bake it..."
-			if(istype(W,/obj/item/reagent_containers/food/snacks/ingredient/))
-				heal_amt += 4
-			else
-				heal_amt += round((F.heal_amt * F.amount)/amount) + 1
-			src.AddComponent(/datum/component/consume/foodheal, src.heal_amt)
+		//We can't actually use realname because tons of foods change their name and don't update it
+		toppings += W.name
+		toppingstext = copytext(html_encode(english_list(toppings)), 1, 512)
+		name = "uncooked [toppingstext] pizza"
+		desc = "A pizza with [toppingstext] toppings. You need to bake it..."
+		if(istype(W,/obj/item/reagent_containers/food/snacks/ingredient/))
+			heal_amt += 4
+		else
+			heal_amt += round((F.heal_amt * F.amount)/amount) + 1
```

Comment:
Unsure what this is trying to achieve. Do we have food stacks with `amount` more than 1? If so, I guess this makes sense, but why add `1` to the average? Does food get better if you put it on a pizza?

Is `amount` here the amount of pizza slices?

Reply:
That's not really my code, just something I moved into a proc